### PR TITLE
Rechecked and update map objects

### PR DESCRIPTION
**Pull request by IvanSavenko**
- Fixed empty messages on some objects
- Marked pickable & sea objects as "blockVisitable" so they are visited from nearby tile instead of moving onto object
- Implemented several objects, namely Derrick, Prospector, Mineral Spring, Watering place (all should 100% match HotA)
- Warehouses should now properly reset instead of being infinitely revisitable
- Rechecked RMG settings & terrains of map objects

### CODE REVIEW

File: hota/Mods/HotA configurable objects/content/config/hotaAncientLamp.json [hunk 1]
```diff
@@ -7,13 +7,14 @@
 			{
 				"name": "Ancient Lamp (allows you to recruit 4-7 Master Genie once)",
 				"rmg" : {
-					"value"		: 5000,
+					"value"		: 500,
```
Comment (misiokles) [inline hota/Mods/HotA configurable objects/content/config/hotaAncientLamp.json]: In documentation, value is 5000. https://heroes.thelazy.net/index.php/Ancient_Lamp

File: hota/Mods/HotA configurable objects/content/config/hotaObservatory.json [hunk 2]
```diff
@@ -7,18 +7,20 @@
 				"sounds" : {
 					"visit" : ["LIGHTHOUSE"]
 				},
-//				"rmg" : {
-//					"value"	: 750,
-//					"rarity" : 100
-//				},
+				"rmg" : {
+					"value"	: 500,
+					"rarity" : 20
```
Comment (misiokles) [inline hota/Mods/HotA configurable objects/content/config/hotaObservatory.json]: ... and "zoneLimit" : 1

File: hota/Mods/HotA configurable objects/content/config/hotaWateringPlace.json [hunk 3]
```diff
@@ -2,30 +2,35 @@
...
-//					"value"		: 500,
-//					"rarity" 	: 50,
-//					"zoneLimit" : 1
-//				},
-				"rewards" :
-				[
+				"aiValue" : 3000,
+				"rmg" : {
+					"value"		: 500,
+					"rarity" 	: 40,
```
Comment (misiokles) [inline hota/Mods/HotA configurable objects/content/config/hotaWateringPlace.json]: "rarity" : 50, in HotA

File: hota/Mods/HotA configurable objects/content/config/hotaAncientLamp.json (hunk 1 above)
Comment (IvanSavenko) [reply hota/Mods/HotA configurable objects/content/config/hotaAncientLamp.json]: Huh, I was sure that it was 500 in docs yesterday... Fixing...

**Comment by IvanSavenko** [comment]
Fixed found issues

**Pull request merged**